### ISSUE TITLE
Automate creation of documentation tasks

### ISSUE THREAD

**Issue by amarthadan** [api3dao/airnode#1235]
I would imagine it working this way:
* Once the issue implementation is finished (PR is opened, the issue is moved to the `In Review` column) we manually add a `documentation` label to that issue/PR, saying that this change needs to be documented
* This (the label addition) will automatically create an issue in the docs repo with a description linking the original issue and PR
* Any additional documentation instructions should be written to this new issue
* The new issue will be automatically moved to the `TODO` column and added to the current iteration and milestone

**Comment by bbenligiray** [comment]
Good point, but keeping docs records separate is neat. What if the docs issue was created as soon as the documentation label has been added? (and you're supposed to add it early on)

**Comment by amarthadan** [comment]
I personally feel like the `Documentation` column in Jira doesn't work very well for us. Writing documentation is often a separate task that can go through the columns as development tasks do. Having just one column for that says nothing about the progress, blockers, whether you're waiting for a review, etc. I think having a separate flow for it makes sense. Also, it kinda makes sense because documentation has a different repository. Otherwise, you would be creating a PR there but referencing basically an already closed issue on the Airnode repo.

As for adding a `documentation` label sooner, sure, why not. I decided not to create the documentation tasks right away (as I wrote in the Slack post) because the point of tasks in the `TODO` column should be that one can take them and work on them. But unless the implementation is done you can't really write the documentation. So, the `documentation` label should be added (and the documentation task is created) once one working on it would be able to write the documentation.

I would very much like to hear what @wkande thinks about this as this flow would be mainly for him.

**Comment by wkande** [comment]
I do like that the documentation can progress in a separate workflow. This helps me follow progress on documentation written by others as well as myself. The question is when will the doc issue be created. You are suggesting once the dev issue moves to "review" it triggers a api3-docs repo issue by adding a documentation label. For those that wish to start earlier could they add the documentation label earlier (manually) by choice which in turn also starts the docs workflow. Of course the automated process will not fire since it is already done.

**Comment by amarthadan** [comment]
Maybe I didn't describe it correctly. I don't want to create a docs repo issue automatically on issue movement but on the manual addition of the `documentation` label. It can't be bound to the movement because we have issues that don't need the documentation. So, the person working on the issue has to add the label manually starting the automation process. And yes, I would do it so the label can be added at any stage and it would kick off the process.

**Comment by wkande** [comment]
Sounds like the perfect solution to me.

**Comment by dcroote** [comment]
I personally liked having the Documentation column as it forced the developer to consider whether documentation was needed and if so, incentivized getting it done in order to complete the task. For tasks not needing documentation, it was easy enough to just comment "documentation isn't needed" and move the task.

I suppose one way to check that all documentation is complete before a release under @amarthadan's plan is to switch to Table view and ensure that there are at least 2 linked PRs (1 airnode, 1 docs) for any tasks with the "documentation" label.

![image](https://user-images.githubusercontent.com/13989647/173981514-a687c6b8-8a63-4e31-adba-9f61538d6972.png)

**Comment by amarthadan** [comment]
I chose pretty much the same approach as for https://github.com/api3dao/airnode/issues/1232
Tested with https://github.com/api3dao/airnode/pull/1200 which worked.
You can check the code [here](https://github.com/amarthadan/github-board-slack-notifications/commit/429d505a3fac6119824c66874606e9f4d21f2b67) but again, more polishing and proper integration to our infra will be done in a separate issue once I see it's working.

Closing.
